perception_eval/perception_eval/tool/perception_analyzer3dfield.py

```diff
@@ -0,0 +1,925 @@
+        """
+        Initialize a PerceptionFieldAxis object.
+
+        Args:
+            type (str): The type of the axis.
+            data_label (str): The label for the data.
+            name (str, optional): The name of the axis. Defaults to "".
+        """
+        # Data parameters
+        self.type: str = type
+        self.data_label: str = data_label
+        # Plot parameters
+        self.name: str = name
+        self.plot_scale: float = 1.0
+        self.unit: str = "none"
+        self.grid_axis: np.ndarray = np.ndarray([0])
+        self.plot_range: Tuple[float, float] = (-1.0, 1.0)
+        self.plot_aspect_ratio: float = 1.0
+
+        if name == "":
+            self.name = data_label
+
+        # set default parameters by type
+        if self.type == "length":
+            self.grid_axis = np.arange(-90.0, 90.0, 10)
+            self.plot_range = (-110.0, 110.0)
+            self.plot_aspect_ratio = 10.0
+            self.unit = "m"
+        elif self.type == "angle":
+            self.grid_axis = np.arange(-180 + 22.5, 180.0, 45) * np.pi / 180.0
+            self.plot_range = (-180.0 - 22.5, 180.0 - 22.5)
+            self.plot_aspect_ratio = 45.0
+            self.plot_scale = 180.0 / np.pi  # convert radian to degree
+            self.unit = "deg"
+        elif self.type == "velocity":
+            self.grid_axis = np.arange(-35.0, 35.0, 5)
+            self.plot_range = (-40.0, 40.0)
+            self.plot_aspect_ratio = 5.0
+            self.unit = "m/s"
+        elif self.type == "acceleration":
+            self.grid_axis = np.arange(-10.0, 10.0, 2)
+            self.plot_range = (-11.0, 11.0)
+            self.plot_aspect_ratio = 2.0
+            self.unit = "m/s^2"
+        elif self.type == "time":
+            self.grid_axis = np.arange(0.0, 5.0, 1)
+            self.plot_range = (0.0, 6.0)
+            self.plot_aspect_ratio = 1.0
+            self.unit = "s"
+
+    def getLabel(self) -> str:
+        """
+        Get the label for the axis.
+
+        Returns:
+            str: The label for the axis.
+        """
+        if self.type == "none":
+            return "x"
+        else:
+            return self.data_label
+
+    def setGridAxis(self, grid_axis: np.ndarray) -> None:
+        """
+        Set the grid axis values.
+
+        Args:
+            grid_axis (np.ndarray): The grid axis values.
+        """
+        if (grid_axis.shape[0] < 2) & (self.type != "none"):
+            raise ValueError("grid_axis must have more than 2 elements.")
+        self.grid_axis = np.sort(grid_axis)
+        if self.type == "angle":
+            self.grid_axis = self.grid_axis * np.pi / 180.0
+
+    def isLoop(self) -> bool:
+        """
+        Check if the axis is a loop.
+
+        Returns:
+            bool: True if the axis is a loop, False otherwise.
+        """
+        if self.type == "angle":
+            return True
+        else:
+            return False
```

Comment:
[IMO] It doesn't have to return as literal, except in cases where other types are included in the future

```suggestion
        return self.type == "angle"
```